src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,211 @@
+package duke;
+
+import duke.command.Command;
+import duke.command.Parser;
+import duke.exception.DukeException;
+import duke.javafx.DialogBox;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.SimpleCliUi;
+import duke.ui.Ui;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+/**
+ * Represents a task list chatbot named Duke.
+ *
+ * @author Terence Chong Guang Jun
+ */
+public class Duke extends Application {
+    /**
+     * Specifies the date format used by <code>Duke</code> for receiving date input.
+     */
+    public static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("ddMMyyyy HHmm");
```

Comment:
maybe it would be better if you made a seperate class to handle converting strings to dates